src/PersonObjects/Sleeve/Work/SleeveBladeburnerWork.ts

```diff
@@ -34,16 +34,34 @@ export class SleeveBladeburnerWork extends Work {
   process(player: IPlayer, sleeve: Sleeve, cycles: number): number {
     if (!player.bladeburner) throw new Error("sleeve doing blade work without being a member");
     this.cyclesWorked += cycles;
+    const actionIdent = player.bladeburner.getActionIdFromTypeAndName(this.actionType, this.actionName);
+    if (!actionIdent) throw new Error(`Error getting ${this.actionName} action`);
+    if (this.actionType === "Contracts") {
+      const action = player.bladeburner.getActionObject(actionIdent);
+      if (!action) throw new Error(`Error getting ${this.actionName} action object`);
+      if (action.count <= 0) {
+        sleeve.stopWork(player);
+        return 0;
+      }
+    }
+
     while (this.cyclesWorked > this.cyclesNeeded(player, sleeve)) {
-      const actionIdent = player.bladeburner.getActionIdFromTypeAndName(this.actionType, this.actionName);
-      if (!actionIdent) throw new Error(`Error getting ${this.actionName} action`);
-      player.bladeburner.completeAction(player, sleeve, actionIdent, false);
+      if (this.actionType === "Contracts") {
+        const action = player.bladeburner.getActionObject(actionIdent);
+        if (!action) throw new Error(`Error getting ${this.actionName} action object`);
+        if (action.count <= 0) {
+          sleeve.stopWork(player);
+          return 0;
+        }
+      }
+      const retValue = player.bladeburner.completeAction(player, sleeve, actionIdent, false);
       let exp: WorkStats | undefined;
       if (this.actionType === "General") {
         exp = GeneralActions[this.actionName]?.exp;
         if (!exp) throw new Error(`Somehow there was no exp for action ${this.actionType} ${this.actionName}`);
         applySleeveGains(player, sleeve, exp, 1);
       }
+      if (retValue.money > 0) player.gainMoney(retValue.money, "sleeves");
```

Comment:
i wonder if this should occur as part of sleeve gains?

Reply:
Preferably not, but putting it elsewhere would require quite a bit of refactoring or passing the `retValue` around